### ISSUE TITLE
To support account management via LDAP

### ISSUE THREAD

**Issue by klorydryk** [wallabag/wallabag#966]
Hi,
I would like to offer this wonderfull tool to my colleagues, in my company. But I can't manage the accounts (creation, deletion) manually. Neither ask to create a new password...
It would be great to support a directory service protocol as LDAP to do that.
Thanks!

**Comment by conditionsofexistence** [comment]
+1

**Comment by smknstd** [comment]
+1

**Comment by stevenroose** [comment]
+1 or just a bump, sadly this one is delayed to 2.0. What's the eta for 2.0?

**Comment by tcitworld** [comment]
> What's the eta for 2.0?

The upcoming weeks should see an alpha version be released. You may be interested by this page : https://www.wallabag.org/pages/roadmap.html

**Comment by stevenroose** [comment]
Thanks! Good to see v2 coming along! Hopefully this feature will still make it to the milestone :)

**Comment by nicosomb** [comment]
see http://symfony.com/blog/new-in-symfony-2-8-ldap-component

**Comment by nebulade** [comment]
Hi, is there any update on the ldap front?

**Comment by j0k3r** [comment]
Well, regarding the features list, I think this feature will more likely coming in the v2.1. What do you think @nicosomb ?

**Comment by nicosomb** [comment]
Yes, we need to plan it for 2.1. We have to be focused on main features to release v2.0 as soon as possible. 
But we don't forget this feature.

**Comment by jeromelebleu** [comment]
Hi! As we need it for the [YunoHost app](https://github.com/YunoHost-Apps/wallabag2_ynh), I tried to make something regarding @nicosomb link - thanks for that by the way - but I'm facing into knowledge lake to improve it... To not pollute this issue, I've created a specific one for that here: https://github.com/YunoHost-Apps/wallabag2_ynh/issues/1. Thank you in advance to those who can help!

**Comment by nerzhul** [comment]
i agree that LDAP integration would be a must have. Self hosting nowadays needs centralized auth like LDAP.

**Comment by JimboJoe** [comment]
Hi, do you see any chance to have LDAP integration implemented any time soon?
We would love to welcome Wallabag v2 as an official YunoHost app ;-)

**Comment by tYYGH** [comment]
Hi! If I understand both issues’ comments correctly, both LDAP authentication and proxy-provided identity are now available in Wallabag, but not yet present in a release.
Would it be possible to create a new release to include LDAP authentication? Latest release is from long ago ;-)

**Comment by j0k3r** [comment]
LDAP auth wont be available in the next release The PR wasn't merged but closed.
Proxy-provided will be easier in release after the 2.3.0 maybe 2.3.1/2 I don't know

**Comment by Zwordi** [comment]
Hello,
Any more recent news about that one ?

**Comment by kettbi** [comment]
Hi,
  
Any news?  
  
I know there is a [patch](https://foss-notes.blog.nomagic.uk/2019/01/installing-wallabag-2-3-6-with-ldap-authentication/) (from immae) for the 2.3.6 version, but I don't know if it will be add in the original project or if the patch is available for upper version.   

thx

**Comment by gilir** [comment]
Hello,

I merged the patch from Immae with the main branch, and it seems to work correctly. If you want to test it, I have 2 branches with this code :
- https://github.com/gilir/wallabag/tree/ldap : Merge with stable 2.3.8 release (I tested it successfully)
- https://github.com/gilir/wallabag/tree/gilir/ldap : Merge with master (same code, but not tested, it probably needs to fix the composer configuration).

Configuration for the ldap server are on wallabag/src/Wallabag/UserBundle/Resources/config/ldap.yml
 
You need to run `SYMFONY_ENV=prod composer require --no-update fr3d/ldap-bundle` before `SYMFONY_ENV=prod composer install` to be sure ldap is enable. Since I'm not a PHP/Symphony dev, there is probably a simple fix for this, in composer configuration.

Also, It works only on new user. When you first login, the user is created with the ldap information in the database, which is needed to work properly (login to the web interface, and the OAuth / API). You can probably manually add those information for existing users in the database, but I didn't test it yet.

**Comment by samsam-rolon** [comment]
Hello, 
Any update regarding the official support of LDAP ?

**Comment by bubbl** [comment]
+1

**Comment by nicosomb** [comment]
@gilir I just had a look and `fr3d/ldap-bundle` is no more maintained. Do you know which bundle could replace it?

**Comment by gilir** [comment]
@nicosomb sorry I dont know :-( I didn't look at this issue since my last comment

**Comment by Scot-Survivor** [comment]
Any updates here?
